### ISSUE TITLE
const enum NamedValues gets compiled into an integer instead of a string (release build)

### ISSUE THREAD

**Issue by Organick** [erik-kallen/SaltarelleCompiler#406]
C#

    [NamedValues]
    [Imported]
    enum Size {
        [ScriptName="small"]
        Small
    }
    
    public class Foo {
        public const Size DefaultSize = Size.Small;
        ...
        public SomeMethod() {
            this.SetSize(DefaultSize); // problem is reflected on this line
        }
    }

JavaScript

    this.setSize(0);

You can see that `DefaultSize` got compiled to 0 instead of `'small'` or `Foo.DefaultSize`. One theory is that there's some kind of race condition in the compiler in handling const vs enum NamedValues.

My current work around this is to use `static readonly` instead of `const`.

**Comment by jon-hall** [comment]
Thought I'd try to reproduce this to see if I could find a fix, but performing a release build against the latest Saltarelle NuGet (2.6.2 Runtime/2.6.0 Compiler) doesn't produce the error described for me (building in VS2013).

The C# I compiled is:

```C#
namespace NamedEnumTest
{
	[NamedValues]
	[Imported]
	public enum SizeEnum
	{
		[ScriptName("small")]
		Small
	}

	public class Foo
	{
		public const SizeEnum DefaultSize = SizeEnum.Small;

		public void SetSize(SizeEnum s)
		{
		}

		public void SomeMethod()
		{
			SetSize(DefaultSize);
		}
	}
}
```

And it produced the following Javascript:

```javascript
(function() {
	'use strict';
	var $asm = {};
	global.NamedEnumTest = global.NamedEnumTest || {};
	ss.initAssembly($asm, 'NamedEnumTest');
	////////////////////////////////////////////////////////////////////////////////
	// NamedEnumTest.Foo
	var $NamedEnumTest_Foo = function() {
	};
	$NamedEnumTest_Foo.__typeName = 'NamedEnumTest.Foo';
	global.NamedEnumTest.Foo = $NamedEnumTest_Foo;
	ss.initClass($NamedEnumTest_Foo, $asm, {
		setSize: function(s) {
		},
		someMethod: function() {
			this.setSize($NamedEnumTest_Foo.defaultSize);
		}
	});
	(function() {
		$NamedEnumTest_Foo.defaultSize = 'small';
	})();
})();
```

I'm not sure if the C# I'm compiling (or my build configuration, which is all just default) is missing something to replicate the issue, or if you might be compiling against an older version of Saltarelle (or a different branch)?

**Comment by erik-kallen** [comment]
I believe some bug that might have caused this was fixed in 2.6, but I also know that the fix was incomplete and that the full fix will need to wait for 3.0 (Roslyn) because lack of support in NRefactory.
